Index: src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java	(date 1525768557000)
+++ src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java	(revision )
@@ -26,7 +26,6 @@
 @Service
 public class CallbackInvoker {
 
-    private static final String CALLBACK_RESPONSE_KEY_STATE = "state";
     private static final HashMap<String, JsonNode> EMPTY_DATA_CLASSIFICATION = Maps.newHashMap();
     private final CallbackService callbackService;
     private final CaseTypeService caseTypeService;
@@ -108,10 +107,13 @@
                                                                                caseDetails,
                                                                                deduceDefaultClassificationForExistingFields(caseType, caseDetails));
             }
-            final Optional<String> newCaseState = filterCaseState(callbackResponse.getData());
-            newCaseState.ifPresent(caseDetails::setState);
-            return newCaseState;
         }
+
+        if (callbackResponse.getState() != null) {
+            caseDetails.setState(callbackResponse.getState());
+            return Optional.of(callbackResponse.getState());
+        }
+
         return Optional.empty();
     }
 
@@ -141,10 +143,4 @@
                                                                                                                      newHashMap()));
         caseDetails.setDataClassification(defaultSecurityClassifications);
     }
-
-    Optional<String> filterCaseState(final Map<String, JsonNode> data) {
-        final Optional<JsonNode> jsonNode = ofNullable(data.get(CALLBACK_RESPONSE_KEY_STATE));
-        jsonNode.ifPresent(value -> data.remove(CALLBACK_RESPONSE_KEY_STATE));
-        return jsonNode.flatMap(value -> value.isTextual() ? Optional.of(value.textValue()) : Optional.empty());
-    }
 }
Index: src/main/java/uk/gov/hmcts/ccd/domain/model/callbacks/CallbackResponse.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/uk/gov/hmcts/ccd/domain/model/callbacks/CallbackResponse.java	(date 1525768557000)
+++ src/main/java/uk/gov/hmcts/ccd/domain/model/callbacks/CallbackResponse.java	(revision )
@@ -21,6 +21,7 @@
     private SecurityClassification securityClassification;
     private List<String> errors;
     private List<String> warnings;
+    private String state;
 
     public Map<String, JsonNode> getData() {
         return data;
@@ -68,6 +69,14 @@
         this.securityClassification = securityClassification;
     }
 
+    public String getState() {
+        return state;
+    }
+
+    public void setState(String state) {
+        this.state = state;
+    }
+
     @JsonIgnore
     @Override
     public String toString() {
